evalml/tests/pipeline_tests/test_pipeline_utils.py

```diff
@@ -40,465 +40,204 @@
-                expected_components = (
-                    [DropNullColumns, Imputer] + delayed_features + estimator_components
-                )
-            pipeline.component_graph.compute_order == [
-                component.name for component in expected_components
-            ]
+    return _get_test_data_from_configuration
 
 
+@pytest.mark.parametrize("lognormal_distribution", [True, False])
 @pytest.mark.parametrize("input_type", ["pd", "ww"])
 @pytest.mark.parametrize("problem_type", ProblemTypes.all_problem_types)
-def test_make_pipeline_text_columns(input_type, problem_type):
-    X = pd.DataFrame(
-        {
-            "numerical": [1, 2, 3, 1, 2],
-            "categorical": ["a", "b", "a", "c", "c"],
-            "text": [
-                "string one",
-                "another",
-                "text for a column, this should be a text column!!",
-                "text string",
-                "hello world",
-            ],
-        }
-    )
-    y = pd.Series([0, 0, 1, 1, 0])
-    if input_type == "ww":
-        X.ww.init()
-        y = ww.init_series(y)
-    estimators = get_estimators(problem_type=problem_type)
-
-    pipeline_class = _get_pipeline_base_class(problem_type)
-    if problem_type == ProblemTypes.MULTICLASS:
-        y = pd.Series([0, 2, 1, 2])
-
-    for estimator_class in estimators:
-        if problem_type in estimator_class.supported_problem_types:
-            parameters = {}
-            if is_time_series(problem_type):
-                parameters = {
-                    "pipeline": {"date_index": None, "gap": 1, "max_delay": 1},
-                    "Time Series Baseline Estimator": {
-                        "date_index": None,
-                        "gap": 1,
-                        "max_delay": 1,
-                    },
-                }
-
-            pipeline = make_pipeline(X, y, estimator_class, problem_type, parameters)
-            assert isinstance(pipeline, pipeline_class)
-            delayed_features = []
-            if is_time_series(problem_type):
-                delayed_features = [DelayedFeatureTransformer]
-            if estimator_class.model_family == ModelFamily.LINEAR_MODEL:
-                estimator_components = [OneHotEncoder, StandardScaler, estimator_class]
-            elif estimator_class.model_family == ModelFamily.CATBOOST:
-                estimator_components = [estimator_class]
-            else:
-                estimator_components = [OneHotEncoder, estimator_class]
-            if estimator_class.model_family == ModelFamily.ARIMA:
-                expected_components = [TextFeaturizer, Imputer] + estimator_components
-            else:
-                expected_components = (
-                    [TextFeaturizer, Imputer] + delayed_features + estimator_components
-                )
-            pipeline.component_graph.compute_order == [
-                component.name for component in expected_components
-            ]
-
+@pytest.mark.parametrize(
```

Comment:
This is awesome! Non-blocking food-for-thought comment hehe but did you consider including the expected answer in the parametrization rather than computing the answer within the test?

Pro: It'd be much easier to add future test cases that are independent of the other ones. I remember having to "fight" with the answer-generation-logic when I was  adding test cases for URL and Email featurizers. 

Con: There are a lot of test cases so it's a lot of data to specify. There will also be some duplicate answers across these tests.

I think this is great as-is but just wanted to put that out there.

Reply:
Gosh, I love this idea a lot. The reason I decided to go with logic, though, is because the estimator also affects what the pipeline looks like, and we currently iterate through each of the estimators in the test. There's room to argue whether or not it's really necessary to loop through all of the estimators as we do or if we should just write out test cases for the special estimators such as CatBoost which can handle categorical cols natively and thus don't require a OneHotEncoder... but maybe this will be worth another iteration of refactoring. 

I do agree though, I had to fight and wrangle the logic quite some bit to get all of these tests to pass, and if we add more logic, I'm not sure how well everything will continue to hold up 😂